nixops/backends/none.py

```diff
@@ -6,12 +6,15 @@
 from nixops.util import attr_property, create_key_pair
 import nixops.resources
 
+from typing import Optional
+
 
 class NoneDefinition(MachineDefinition):
     """Definition of a trivial machine."""
 
     _target_host: str
     _public_ipv4: Optional[str]
+    _provision_ssh_key: bool
```

Comment:
Let's move this to MachineDefinition's base implementation, and from there plugins need to handle it (and failing to handle it can be considered a bug.)